src/utils.py

```diff
@@ -0,0 +1,406 @@
+import itertools
+import os
+from pathlib import Path
+from typing import Literal
+
+import cftime
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rioxarray as rxr
+import xarray as xr
+from ochanticipy import CodAB, create_country_config
+from rasterio.enums import Resampling
+from shapely import box
+from tqdm import tqdm
+
+DATA_DIR = Path(os.getenv("AA_DATA_DIR"))
+
+
+def load_codab_aoi() -> gpd.GeoDataFrame:
+    """Loads CODAB of AOI"""
+    load_dir = DATA_DIR / "public/processed/sah/cod_ab"
+    filename = "bfa-ner-tcd_adm2_codab_aoi.shp.zip"
+    return gpd.read_file(f"zip://{load_dir / filename}")
+
+
+def process_clip_cod_to_aoi():
+    """Clip CODAB to area of interest for frameworks. In Burkina and Chad,
+    based on specific admin1s. In Niger, is everything south of 17N.
+    Returns
+    -------
+
+    """
+    cod = load_codab_all()
+    bfa_adm1_codes = ["BF46", "BF49", "BF54", "BF56"]
+    bfa_clip = cod[cod["ADM1_CODE"].isin(bfa_adm1_codes)]
+    tcd_adm1_codes = ["TD01", "TD06", "TD07", "TD17", "TD19"]
+    tcd_clip = cod[cod["ADM1_CODE"].isin(tcd_adm1_codes)]
+    ner_clip = cod[cod["ADM0_CODE"] == "NER"].clip(box(-180, -90, 180, 17))
+    cod_clip = pd.concat([bfa_clip, tcd_clip, ner_clip], ignore_index=True)
+    save_dir = DATA_DIR / "public/processed/sah/cod_ab"
+    filename = "bfa-ner-tcd_adm2_codab_aoi.shp.zip"
+    cod_clip.to_file(save_dir / filename)
+
+
+def load_asap_sos_eos() -> xr.DataArray:
+    """Loads ASAP start/end/senescence of season for countries of interest.
+
+    Returns
+    -------
+    xr.DataArray
+    """
+    load_dir = DATA_DIR / "public/processed/sah/asap"
+    da_ss = []
+    for season in [1, 2]:
+        da_ses = []
+        for s_e in ["s", "e", "sen"]:
+            filename = f"pheno{s_e}{season}_v03_sah.tif"
+            da_se = rxr.open_rasterio(load_dir / filename).astype("uint8")
+            da_se["s_e"] = s_e
+            da_ses.append(da_se)
+        da_s = xr.concat(da_ses, dim="s_e")
+        da_s["season"] = season
+        da_ss.append(da_s)
+    da = xr.concat(da_ss, dim="season")
+    da = da.squeeze(drop=True)
+    return da
+
+
+def load_iri() -> xr.DataArray:
+    """Loads IRI low tercile probability over AOI
+
+    Returns
+    -------
+    xr.DataArray
+    """
+    load_dir = DATA_DIR / "private/processed/sah/iri"
```

Comment:
Yeah it's just what [AnticiPy does](https://ocha-anticipy.readthedocs.io/en/latest/datasources/iri_seasonal_forecast.html#usage), because you need to make an account (that presumably gets authenticated somehow by IRI) to get an API key.